src/engine/renderer/glsl_source/reliefMapping_fp.glsl

```diff
@@ -268,10 +279,10 @@ float RayIntersectDisplaceMap(vec2 dp, vec2 ds, sampler2D normalMap)
 	{
 		depth += size;
 
-		vec4 t = texture2D(normalMap, dp + ds * depth);
+		float t = 1.0 - texture2D(normalMap, texCoords + ds * depth).a;
```

Comment:
Why is the 'one minus' introduced?

Reply:
That's because parallax digs hole into the texture from the top and top is 1.0:

```
 1 ____      1-0 ____
  |    \        /
  |     \      /
.5|      \____/ 1-.5
  |
  |
 0|
```

The height the must be the floor, basically the heightmap parallax is a depth map.

A bias may then be introduced between `1.0` and `texture2D(normalMap, texCoords + ds * depth).a`

Notice the `-` in the `-offsetScale` that is just a `* -1.0` which makes the depth a negative number that may be substracted to `1.0`.

Reply:
The miss of that “_one minus_” was the cause of the reverted parallax (and maybe wrong offset).

That is the proper fix.

Reply:
It was confusing since the commit was labeled "refactoring" and all other parts moved stuff around without behavioral changes.

Reply:
Oh right, that slipped in with a fixup, I extracted it to its own commit.